src/libasr/pass/intrinsic_array_function_registry.h

```diff
@@ -1316,6 +1318,265 @@ namespace Cshift {
+            value = eval_Eoshift(al, loc, ret_type, m_args, diag);
+        }
+        return make_IntrinsicArrayFunction_t_util(al, loc,
+            static_cast<int64_t>(IntrinsicArrayFunctions::Eoshift),
+            m_args.p, m_args.n, overload_id, ret_type, value);
+    }
+
+    static inline ASR::expr_t* instantiate_Eoshift(Allocator &al,
+            const Location &loc, SymbolTable *scope,
+            Vec<ASR::ttype_t*> &arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t> &m_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_eoshift");
+        fill_func_arg("array", duplicate_type_with_empty_dims(al, arg_types[0]));
+        fill_func_arg("shift", arg_types[1]);
+        fill_func_arg("boundary", arg_types[2]);
+        ASR::ttype_t* return_type_ = return_type;
+        /*
+            Eoshift(array, shift, dim)
+            int i = 0
+            for j = shift, size(array)
+                result[i] = array[j]
+                i = i + 1
+            end for
+            for j = 1, shift
+                result[i] = array[j]
+                i = i + 1
+            end for
+
+            if (k shift= 0) then
+                i = size(array)
+                i = i - shift
+                i = i + 1
+            else
+                i = 1
+            end if
+
+            do j = 1, shift
+                result(i) = boundary
+                i = i + 1
+            end do
+        */
+        if( !ASRUtils::is_fixed_size_array(return_type) ) {
+            bool is_allocatable = ASRUtils::is_allocatable(return_type);
+            Vec<ASR::dimension_t> empty_dims;
+            empty_dims.reserve(al, 2);
+            for( int idim = 0; idim < 2; idim++ ) {
+                ASR::dimension_t empty_dim;
+                empty_dim.loc = loc;
+                empty_dim.m_start = nullptr;
+                empty_dim.m_length = nullptr;
+                empty_dims.push_back(al, empty_dim);
+            }
+            return_type_ = ASRUtils::make_Array_t_util(al, loc,
+                ASRUtils::extract_type(return_type_), empty_dims.p, empty_dims.size());
+            if( is_allocatable ) {
+                return_type_ = ASRUtils::TYPE(ASR::make_Allocatable_t(al, loc, return_type_));
+            }
+        }
+        ASR::expr_t *result = declare("result", return_type_, Out);
+        args.push_back(al, result);
+        ASR::expr_t *i = declare("i", int32, Local);
+        ASR::expr_t *j = declare("j", int32, Local);
+        ASR::expr_t* abs_shift = declare("z", int32, Local);
+        ASR::expr_t* abs_shift_val = declare("k", int32, Local);
+        ASR::expr_t* shift_val = declare("shift_val", int32, Local);;
+        ASR::expr_t* boundary = args[2];
+
+        body.push_back(al, b.Assignment(shift_val, b.i2i_t(args[1], int32)));
+        body.push_back(al, b.Assignment(abs_shift, b.i2i_t(shift_val, int32)));
+        body.push_back(al, b.Assignment(abs_shift_val, b.i2i_t(shift_val, int32)));
```

Comment:
```suggestion
        body.push_back(al, b.Assignment(shift_val, args[1]));
        body.push_back(al, b.Assignment(abs_shift, shift_val));
        body.push_back(al, b.Assignment(abs_shift_val, shift_val));
```
Do we need to IntegerToInteger casting here?